client/network/src/protocol.rs

```diff
@@ -413,6 +462,23 @@ impl<B: BlockT> NetworkBehaviour for Protocol<B> {
 			return Poll::Ready(NetworkBehaviourAction::GenerateEvent(message))
 		}
 
+		while let Poll::Ready(Some(validation_result)) =
+			self.sync_substream_validations.poll_next_unpin(cx)
+		{
+			match validation_result {
+				Ok((peer, index)) => {
+					self.behaviour.accept_inbound_substream(peer, index);
+				},
+				Err(index) => {
+					log::debug!(
+						target: "sub-libp2p",
+						"`SyncingEngine` rejected stream"
+					);
+					self.behaviour.peerset_report_reject(index);
```

Comment:
May be introduce `self.behaviour.reject_inbound_substream(index)`, supplementing `accept_inbound_substream`, to not confuse readers with `peerset` reference here?